File: web/src/rulesets/gfm.ts
import alertRule from "../block/alertRule";
import blockQuoteRule from "../block/blockQuoteRule";
import codeBlockRule from "../block/codeBlockRule";
import codeFenceRule from "../block/codeFenceRule";
import contentRule from "../block/contentRule";
import footnoteReferenceRule from "../block/footnoteReferenceRule";
import headingRule from "../block/headingRule";
import headingUnderlineRule from "../block/headingUnderlineRule";
import htmlBlockRule from "../block/htmlBlockRule";
import indentRule from "../block/indentRule";
import linkReferenceRule from "../block/linkReferenceRule";
import listBulletedRule from "../block/listBulletedRule";
import listItemRule from "../block/listItemRule";
import listOrderedRule from "../block/listOrderedRule";
import listTaskItemRule from "../block/listTaskItemRule";
import paragraphRule from "../block/paragraphRule";
import tableRule from "../block/tableRule";
import thematicBreakRule from "../block/thematicBreakRule";
import autolinkRule from "../inline/autolinkRule";
import codeSpanRule from "../inline/codeSpanRule";
import emphasisRule from "../inline/emphasisRule";
import extendedAutolinkRule from "../inline/extendedAutolinkRule";
import footnoteRule from "../inline/footnoteRule";
import hardBreakRule from "../inline/hardBreakRule";
import htmlSpanRule from "../inline/htmlSpanRule";
import linkRule from "../inline/linkRule";
import strikethroughRule from "../inline/strikethroughRule";
import textRule from "../inline/textRule";
import type RuleSet from "../types/RuleSet";

/**
 * The rules from [CommonMark](https://spec.commonmark.org/) plus the [GitHub
 * Flavored Markdown](https://github.github.com/gfm/) rules.
 */
const gfm: RuleSet = {
	blocks: [
		indentRule,
		headingRule,
		headingUnderlineRule,
		thematicBreakRule,
		alertRule,
		blockQuoteRule,
		listOrderedRule,
		listBulletedRule,
		listItemRule,
		listTaskItemRule,
		footnoteReferenceRule,
		codeBlockRule,
		codeFenceRule,
		htmlBlockRule,
		linkReferenceRule,
		tableRule,
		paragraphRule,
		contentRule,
	],
	inlines: [
		autolinkRule,
		extendedAutolinkRule,
		htmlSpanRule,
		codeSpanRule,
		emphasisRule,
		strikethroughRule,
		footnoteRule,
		linkRule,
		hardBreakRule,
		textRule,
	],
};

export default gfm;

File: dotnet/Allmark/Rulesets/Gfm.cs
namespace Allmark.Rulesets;

using Allmark.Block;
using Allmark.Inline;
using Allmark.Render;
using Allmark.Types;

/// <summary>
/// The rules from [CommonMark](https://spec.commonmark.org/) plus [GitHub
/// Flavored Markdown](https://github.github.com/gfm/) rules.
/// </summary>
public static class Gfm
{
    public static RuleSet RuleSet => new RuleSet
    {
        Blocks = new Dictionary<string, BlockRule>
        {
            [IndentRule.Create().Name] = IndentRule.Create(),
            [HeadingRule.Create().Name] = HeadingRule.Create(),
            [HeadingUnderlineRule.Create().Name] = HeadingUnderlineRule.Create(),
            [ThematicBreakRule.Create().Name] = ThematicBreakRule.Create(),
            [AlertRule.Create().Name] = AlertRule.Create(),
            [BlockQuoteRule.Create().Name] = BlockQuoteRule.Create(),
            [ListOrderedRule.Create().Name] = ListOrderedRule.Create(),
            [ListBulletedRule.Create().Name] = ListBulletedRule.Create(),
            [ListItemRule.Create().Name] = ListItemRule.Create(),
            [ListTaskItemRule.Create().Name] = ListTaskItemRule.Create(),
            [FootnoteReferenceRule.Create().Name] = FootnoteReferenceRule.Create(),
            [CodeBlockRule.Create().Name] = CodeBlockRule.Create(),
            [CodeFenceRule.Create().Name] = CodeFenceRule.Create(),
            [HtmlBlockRule.Create().Name] = HtmlBlockRule.Create(),
            [LinkReferenceRule.Create().Name] = LinkReferenceRule.Create(),
            [TableRule.Create().Name] = TableRule.Create(),
            [ParagraphRule.Create().Name] = ParagraphRule.Create(),
            [ContentRule.Create().Name] = ContentRule.Create(),
        },
        Inlines = new Dictionary<string, InlineRule>
        {
            [AutolinkRule.Create().Name] = AutolinkRule.Create(),
            [ExtendedAutolinkRule.Create().Name] = ExtendedAutolinkRule.Create(),
            [HtmlSpanRule.Create().Name] = HtmlSpanRule.Create(),
            [CodeSpanRule.Create().Name] = CodeSpanRule.Create(),
            [EmphasisRule.Create().Name] = EmphasisRule.Create(),
            [StrikethroughRule.Create().Name] = StrikethroughRule.Create(),
            [FootnoteRule.Create().Name] = FootnoteRule.Create(),
            [LinkRule.Create().Name] = LinkRule.Create(),
            [HardBreakRule.Create().Name] = HardBreakRule.Create(),
            [LineBreakRule.Create().Name] = LineBreakRule.Create(),
            [TextRule.Create().Name] = TextRule.Create(),
        },
    };
}
